app.js

```diff
@@ -56,32 +66,45 @@ app.post('/api/shorturl', (req,res) => {
             if (!codeGenerator.checkValidShortCode(shortCode)){
                 sendError(res, "Short code must only be alphanumeric characters");
             //Make sure the short code doesn't exist in DB already
-            } else if(!checkShortCodeIsUnique(shortCode)) {
-                sendError(res, "Short code already in use");
-            } else {    
-                //Check host is valid
-                host = new URL(urlToShorten).host
-                dns.lookup(host, (err) => {
-                    if (err) {
-                        sendError(res, "Bad URL");
+            } else {
+                dbUtil.getShortCode(shortCode).then(
+                //Existing Short Code was found
+                () => {
+                    sendError(res, "Short code already in use");
```

Comment:
The way I've described the translation process from `.then` to `await` might make it seem like you'd need to convert code with nested `.then`s into a bunch of `try/catch` blocks in sequence, one per `.then`. But, if all the errors that might happen during each one of those are essentially equivalent, and you'd handle them the same way, then you can combine those into one `try/catch` with all the main steps in the `try`.

Except, if you use errors to communicate something other than "disaster, something totally unexpected happened", then the caller needs to have separate try/catch for that part ( because it *expects* to maybe get the error when deciding what to do). Just makes the calling code less clean.

Maybe ignore this specific piece of feedback at first and work through the rest, and then it'll be clear how this makes callers a bit clunky. Then decide whether to take another step to de-clunk it. :)